src/components/pages/dashboard/panels/telemetry/telemetryPanel.js

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import React, { Component } from 'react';
+import { Observable } from 'rxjs';
+// import update from 'immutability-helper';
+import 'tsiclient';
+
+import { TelemetryService } from 'services';
+import { Indicator } from 'components/shared';
+import {
+  Panel,
+  PanelHeader,
+  PanelContent,
+  PanelOverlay
+} from 'components/pages/dashboard/panel';
+import { joinClasses } from 'utilities';
+
+import './telemetryPanel.css';
+// TODO: find a way to import without the relative path
+import '../../../../../../node_modules/tsiclient/tsiclient.css';
+
+const telemetryChartId = 'device-telemetry-chart';
+
+const chartColors = [
+  '#01B8AA',
+  '#F2C80F',
+  '#E81123',
+  '#3599B8',
+  '#33669A',
+  '#26FFDE',
+  '#E0E7EE',
+  '#FDA954',
+  '#FD625E',
+  '#FF4EC2',
+  '#FFEE91'
+];
+
+export class TelemetryPanel extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isPending: true,
+      telemetry: {},
+      telemetryKeys: [],
+      telemetryKey: ''
+    };
+
+    // Initialize chart client
+    this.tsiClient = new window.TsiClient();
+  }
+
+  componentDidMount() {
+    this.getData();
+
+    // Create line chart
+    this.lineChart = new this.tsiClient.ux.LineChart(document.getElementById(telemetryChartId));
+  }
+
+  componentWillUnmount() {
+    if (this.subscription) this.subscription.unsubscribe();
+  }
+
+  componentWillUpdate(_, { telemetry, telemetryKey}) {
+    if (telemetry && telemetryKey && telemetry[telemetryKey]) {
+      const datum = Object.keys(telemetry[telemetryKey]).map(deviceId => ({
+        [deviceId]: telemetry[telemetryKey][deviceId]
+      }));
+      const noAnimate = telemetryKey === this.state.telemetryKey;
+      // Set a timeout to allow the panel height to be calculated before updating the graph
+      setTimeout(() => {
+        this.lineChart.render(
+          datum,
+          {
+            grid: false,
+            legend: 'compact',
+            noAnimate, // If the telemetryKey changes, animate
+            tooltip: true,
+            yAxisState: 'shared' // Default to all values being on the same axis
+          },
+          chartColors.map(color => ({ color }))
+        );
+      }, 10);
```

Comment:
suggest making 10 a const

Reply:
I'll take this under consideration but this call will later be wrapped in a component to hide the tsiChart config data. Since this isn't strictly speaking an app constant I would lean away from making an app constant.